src/config.rs

```diff
@@ -1903,24 +1911,32 @@ impl ConfigHandler {
 }
 
 impl ConfigHandler {
-    /// Register the local config to pd
+    /// Register the local config to pd and get the latest
+    /// version and config
     pub fn create(
         id: String,
         pd_client: Arc<impl PdClient>,
-        cfg: TiKvConfig,
+        local_config: TiKvConfig,
     ) -> CfgResult<(configpb::Version, TiKvConfig)> {
-        let cfg = toml::to_string(&cfg)?;
+        let cfg = toml::to_string(&local_config)?;
         let version = configpb::Version::default();
         let mut resp = pd_client.register_config(id, version, cfg)?;
         match resp.get_status().get_code() {
             StatusCode::Ok | StatusCode::WrongVersion => {
-                let cfg: TiKvConfig = toml::from_str(resp.get_config())?;
-                Ok((resp.take_version(), cfg))
-            }
-            code => {
-                debug!("failed to register config"; "status" => ?code);
-                Err(format!("{:?}", resp).into())
+                let mut incoming: TiKvConfig = toml::from_str(resp.get_config())?;
+                let mut version = resp.take_version();
+                if let Err(e) = incoming.validate() {
+                    warn!(
+                        "config from pd is invalid, fallback to local config";
+                        "version" => ?version,
+                        "error" => ?e,
+                    );
+                    version = configpb::Version::default();
+                    incoming = local_config;
+                }
+                Ok((version, incoming))
             }
+            _ => Err(format!("failed to register config, response: {:?}", resp).into()),
```

Comment:
Is it possible that pd is temporarily unavailable and TiKV will not start?

Reply:
You are right, in this situation TiKV will not start and user had to retry, which not just because of this but also TiKV can not `PutStore` to PD.